### PR TITLE
support init dataset from ndarrays

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

**Comment by yangw1234** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2115/ jenkins pass

File: pyzoo/zoo/tfpark/estimator.py [hunk 1]
```diff
@@ -62,15 +62,13 @@ def add_train_op(model_fn, features, labels, mode, params, config, optimizer):
                 " that do not support gradients, between variables %s and loss %s." %
                 ([str(v) for _, v in grads_and_vars], spec.loss))
 
-        train_op = spec.optimizer.apply_gradients(grads_and_vars,
-                                                  global_step=tf.train.get_global_step())
+        train_op = optimizer.apply_gradients(grads_and_vars,
+                                             global_step=tf.train.get_global_step())
```
Comment (jason-dai) [inline pyzoo/zoo/tfpark/estimator.py]: should be `train_op = optimizer.minimize(spec.loss, global_step=tf.train.get_global_step())`, no?

File: pyzoo/test/zoo/tfpark/test_tfpark_estimator.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def input_fn(mode):
                                              features=(tf.float32, [10]),
                                              labels=(tf.int32, []),
                                              batch_size=4)
+            elif mode == tf.estimator.ModeKeys.EVAL:
```
Comment (jason-dai) [inline pyzoo/test/zoo/tfpark/test_tfpark_estimator.py]: this condition is redundant - the if statement above already checks whether it is EVAL mode

Comment (yangw1234) [reply pyzoo/test/zoo/tfpark/test_tfpark_estimator.py]: yeah, I should have deleted the "or mode == tf.estimator.ModeKeys.Eval" above

File: pyzoo/zoo/tfpark/estimator.py (hunk 1 above)
Comment (yangw1234) [reply pyzoo/zoo/tfpark/estimator.py]: `minimize(loss)` is equivalent to first `compute_gradient(loss)` then `apply_gradients`. It got to this way because I used to want to record the gradient.  I'll change to the `minimize` version for simpilicity.
